src/fparser/two/utils.py

```diff
@@ -1179,35 +1175,52 @@ class EndStmtBase(StmtBase):
     """
     @staticmethod
     def match(stmt_type, stmt_name, string, require_stmt_type=False):
-        start = string[:3].upper()
-        if start != 'END':
+        if string[:3].upper() != 'END':
             return
+
+        # Get the full string minus the leading "END" word.
         line = string[3:].lstrip()
-        start = line[:len(stmt_type)].upper()
-        if start:
-            if start.replace(' ', '') != stmt_type.replace(' ', ''):
```

Comment:
I've discovered that this line (the replace in particular) is not tested. I'll happily take a TODO on adding a test for that (PR incoming)

Reply:
Ref: #181